sos/report/plugins/openstack_database.py

```diff
@@ -0,0 +1,83 @@
+# Copyright (C) 2021 Red Hat, Inc., Jake Hunsaker <jhunsake@redhat.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+import re
+
+
+from sos.report.plugins import Plugin, RedHatPlugin
+
+
+class OpenStackDatabase(Plugin):
+
+    short_desc = 'Openstack Database Information'
+    plugin_name = 'openstack_database'
+    profiles = ('openstack', 'openstack_controller')
+
+    option_list = [
+        ('dump', 'Dump select databases to a SQL file', 'slow', False),
+        ('dumpall', 'Dump ALL databases to a SQL file', 'slow', False)
+    ]
+
+    databases = [
+        'cinder',
+        'glance',
+        'heat',
+        'ironic',
+        'keystone',
+        'mistral',
+        '(.*)?neutron',
+        'nova.*'
+    ]
+
+    def setup(self):
+
+        # determine if we're running databases on the host or in a container
+        in_container = self.container_exists('galera-bundle-.*')
+        cname = None
```

Comment:
Isn't it easier to call directly here:

    cname = self.get_container_by_name('galera-bundle-.*')
    in_container = (cname != None)

One iteration over containers is prevented.

Reply:
Ack, valid point.